### PR TITLE
Added backbone model representations for UserEvaluation and Constraint

### LINKED ISSUE #417: Convert UserEvaluation into Backbone.js format
Same as #413 but for UserEvaluation

### LINKED ISSUE #418: Convert Constraint into Backbone.js Model
Same as #413 but for Constraint

### PR DESCRIPTION
**Pull request by YesugenB**
😳

### CODE REVIEW

File: leaf-ui/rappid-extensions/BackboneModels.js [hunk 1]
```diff
@@ -1 +1,30 @@
...
+        this.constraintType = options.constraintType;
+        this.constraintSrcID = options.constraintSrcID;
+        this.constraintSrcEB = options.constraintSrcEB;
+        this.constraintDestID = options.constraintDestID;
+        this.constraintDestEB = options.constraintDestEB;
+        this.absoluteValue = -1;
+    }
+});
+
+//var intentionEval = new UserEvaluationModel({intentionID:intention.nodeID},{absTime:'0'}, {evaluationValue:'(no value)'});
```
Comment (leshghi) [inline leaf-ui/rappid-extensions/BackboneModels.js]: Delete these commented lines

**Review by leshghi — COMMENTED** [review]
This looks good to me from a technical backbone standpoint. Passing it off to @amgrubb so we can discuss any potential structural changes before checking it in to codebase

File: leaf-ui/rappid-extensions/BackboneModels.js [hunk 2]
```diff
@@ -1 +1,27 @@
...
+        this.intentionID = options.intentionID; 
+        this.absTime = options.absTime;
+        this.evaluationValue = options.evaluationValue;
+    }
+});
+
+/** 
+ * Backbone for Constraint 
+ * */
+var ConstraintModel = Backbone.Model.extend({
```
Comment (amgrubb) [inline leaf-ui/rappid-extensions/BackboneModels.js]: I think we are going to change this to be called `ConstraintBBM`.

File: leaf-ui/rappid-extensions/BackboneModels.js [hunk 3]
```diff
@@ -1 +1,27 @@
...
+    initialize: function(options){ //Named arguments and passes in parameter
+        this.constraintType = options.constraintType;
+        this.constraintSrcID = options.constraintSrcID;
+        this.constraintSrcEB = options.constraintSrcEB;
+        this.constraintDestID = options.constraintDestID;
+        this.constraintDestEB = options.constraintDestEB;
+        this.absoluteValue = -1;
```
Comment (amgrubb) [inline leaf-ui/rappid-extensions/BackboneModels.js]: Is `-1` the default value or can we not add an absolute value?

I would like to make the `absoluteValue` to be an optional parameter that defaults to -1.

File: leaf-ui/rappid-extensions/BackboneModels.js [hunk 4]
```diff
@@ -1 +1,27 @@
...
+});
+
+/** 
+ * Backbone for Constraint 
+ * */
+var ConstraintModel = Backbone.Model.extend({
+    initialize: function(options){ //Named arguments and passes in parameter
+        this.constraintType = options.constraintType;
+        this.constraintSrcID = options.constraintSrcID;
+        this.constraintSrcEB = options.constraintSrcEB;
```
Comment (amgrubb) [inline leaf-ui/rappid-extensions/BackboneModels.js]: We can remove constraint from each of these. Then write out `srcID`. 
We are going to change the `EB` to be called `RefTP`, for Reference Time Point, as in `srcRefTP`.

File: leaf-ui/rappid-extensions/BackboneModels.js [hunk 5]
```diff
@@ -1 +1,27 @@
-/** This file contains backbone model representations of the original model objects - WIP */
\ No newline at end of file
+/** This file contains backbone model representations of the original model objects - WIP */
+
+/** 
+ * Backbone for UserEvaluation 
+ * */ 
+var UserEvaluationModel = Backbone.Model.extend({
```
Comment (amgrubb) [inline leaf-ui/rappid-extensions/BackboneModels.js]: `userEvaluationBBM`

File: leaf-ui/rappid-extensions/BackboneModels.js [hunk 6]
```diff
@@ -1 +1,27 @@
...
+
+/** 
+ * Backbone for UserEvaluation 
+ * */ 
+var UserEvaluationModel = Backbone.Model.extend({  
+    //Take in properties
+    initialize: function(options){ //Named arguments and passes in parameter
+        this.intentionID = options.intentionID; 
+        this.absTime = options.absTime;
+        this.evaluationValue = options.evaluationValue;
```
Comment (amgrubb) [inline leaf-ui/rappid-extensions/BackboneModels.js]: Change `this.evaluationValue` to be `this.assignedValue`.
And `this.absTime` to `this.absTP`.

**Review by amgrubb — CHANGES_REQUESTED** [review]
I'm requesting the renaming of items. Let me know if this is not the best time to do this.

File: leaf-ui/rappid-extensions/BackboneModels.js (hunk 5 above)
Comment (amgrubb) [reply leaf-ui/rappid-extensions/BackboneModels.js]: Oops, `UserEvaluationBBM`

File: leaf-ui/rappid-extensions/BackboneModels.js (hunk 3 above)
Comment (amgrubb) [reply leaf-ui/rappid-extensions/BackboneModels.js]: If `type` is `<` or `<=`, it can't have an absolute value associated with it.

**Review by amgrubb — CHANGES_REQUESTED** [review]
Added an additional change.

File: leaf-ui/rappid-extensions/BackboneModels.js (hunk 3 above)
Comment (amgrubb) [reply leaf-ui/rappid-extensions/BackboneModels.js]: Also, can we change `this.absoluteValue` to `this.absTP`.

File: leaf-ui/rappid-extensions/BackboneModels.js (hunk 6 above)
Comment (YesugenB) [reply leaf-ui/rappid-extensions/BackboneModels.js]: >Also, can we change this.absoluteValue to this.absTP.
>And this.absTime to this.absTP.


So will both this.absTime and this.absoluteValue become this.absTP

File: leaf-ui/rappid-extensions/BackboneModels.js [hunk 7]
```diff
@@ -1 +1,37 @@
...
+/** 
+ * Backbone for Constraint 
+ * */
+var ConstraintBBM = Backbone.Model.extend({
+    initialize: function(options){ //Named arguments and passes in parameter
+        this.type = options.type;
+        this.srcID = options.srcID;
+        this.srcRefTP = options.srcRefTP;
+        this.destID = options.destID;
+        this.destEB = options.destEB;
```
Comment (amgrubb) [inline leaf-ui/rappid-extensions/BackboneModels.js:24]: destRefTP

**Review by amgrubb — APPROVED** [review]
@leshghi @YesugenB Please review my two commits.

**Review by leshghi — APPROVED** [review]
Looks good to me just remove the defaults now that absTime is set in initialize

**Pull request merged**